.github/workflows/docker.build.yml

```diff
@@ -7,7 +7,7 @@ on:
     tags:
       - '*'
   pull_request:
-
+    types: ['opened', 'closed', 'reopened', 'synchronize', 'edited']
```

Comment:
you are right. I think removing `edited` event would have it not build for every commit that's probably the only one we don't need i think. Perhaps, we want to keep the others. the default `pull request` `actions` that trigger the workflow are `opened, synchronize, reopened` it is not triggered by the `close` action by default https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

that was the reason i added that list  and the `===approved` because otherwise it creates the PR on `wbaas-deploy` even just from an open PR on UI.

Reply:
Ah ok, I didn't know that `pull_request:` itself already sets up triggers, I was thinking this meant "do nothing on a PR".

For my understanding: why do we need to build and publish images for PRs? Is this just to see if the image builds fine?

Reply:
I also would prefer another way to do this as having `===approved` line written 3 times in the same file is not ideal.

Reply:
> I also would prefer another way to do this as having ===approved line written 3 times in the same file is not ideal.

I'm not entirely sure if we should even open an automated PR on approval because of the following scneario:

- I open a PR to this repo with some changes
- There's some back and forth with changes, I push a few commits
- The PR gets approved
- Now, this action opens a PR against `wbaas-deploy` using the git SHA from my feature branch
- I merge my PR and all of my commits get squashed into `main` creating a new commit. Ideally, I delete my feature branch.
- The SHA that we now want to deploy in `wbaas-deploy` does not exist in the repo anymore as:
  - `main` has different commits
  - my feature branch was deleted

Obviously, this works as the image is on `ghcr` already, but it's probably pretty confusing when you want to check which code is currently running in the cluster?

To me, the easiest way of solving this would having this action run on changes to `main` only, but I might be missing something you used to do with the images built off PR branches. If we just want to check that the image builds, we can also add a second action that does just that for PRs and not for `main`.